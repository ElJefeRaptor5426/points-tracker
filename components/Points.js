import React from 'react';
import { Text } from 'react-native';

export default ({ points }) => (
  <Text>{points}</Text>
)