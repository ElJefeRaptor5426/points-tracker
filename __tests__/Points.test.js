import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'react-native'
import Points from '../components/Points';

Text.displayName = 'Text';

describe('Points component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Points />, {points: 0})
    expect(wrapper).toMatchSnapshot();
  });

  it('should have 1 Text component', () => {
    const wrapper = shallow(<Points points={0}/>);
    expect(wrapper.find('Text').length).toBe(1)
  })

  it('should have text showing the number of points', () => {
    const wrapper = shallow(<Points points={0}/>);
    expect(wrapper.contains(<Text>0</Text>)).toBe(true);
  })
})