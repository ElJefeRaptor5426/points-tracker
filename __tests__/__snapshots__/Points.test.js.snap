// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Points component should render correctly 1`] = `ShallowWrapper {}`;
